provisioner/terraform/resources.go

```diff
@@ -464,7 +470,31 @@ func ConvertResourcesAndParameters(modules []*tfjson.StateModule, rawGraph strin
 		}
 	}
 
-	return resources, parameters, nil
+	// A map is used to ensure we don't have duplicates!
```

Comment:
curious: when is it possible to have duplicates?

Reply:
If a template author did:

```
data "coder_git_auth" "github" {
  id = "github"
}

data "coder_git_auth" "github2" {
  id = "github"
}
```

It would be valid syntax, but we wouldn't want to display auth for the same provider twice.

Reply:
👍 

I'm wondering if we can keep an eye on this in the validation procedure in terraform-provider-coder. It's semantic validation (relationship between resources).

Reply:
I'll do this!